src/project-manager.ts

```diff
@@ -415,21 +532,51 @@ export class ProjectManager implements Disposable {
 }
 
 /**
- * Implementaton of LanguageServiceHost that works with in-memory file system
+ * Implementaton of LanguageServiceHost that works with in-memory file system.
+ * It takes file content from local cache and provides it to TS compiler on demand
+ *
+ * @implements ts.LanguageServiceHost
  */
 export class InMemoryLanguageServiceHost implements ts.LanguageServiceHost {
 
 	complete: boolean;
 
+	/**
+	 * Root URI
+	 */
 	private root: string;
+
+	/**
+	 * Compiler options to use when parsing/analyzing source files.
+	 * We are extracting them from tsconfig.json or jsconfig.json
+	 */
 	private options: ts.CompilerOptions;
+
+	/**
+	 * Local file cache where we looking for file content
+	 */
 	private fs: InMemoryFileSystem;
+
+	/**
+	 * List of files that project consist of (based on tsconfig includes/excludes and wildcards)
+	 */
 	expectedFiles: string[];
 
+	/**
+	 * Current list of files that were implicitly added to project
+	 * (every time when we need to extract data from a file that we havem't touched yet)
+	 */
 	private files: string[];
```

Comment:
paths or URIs? relative to root path?
Do these have to be ordered or could this be a `Set`?